### PR TITLE
added support for passing --defines arguments on command line

### PR DESCRIPTION
**Pull request by reshadi**
This is a fix for this issue #90

### CODE REVIEW

**Comment by reshadi** [comment]
I singed it!

Thanks,
--mehrdad.

________________________________

File: test/simple.js [hunk 1]
```diff
@@ -86,4 +86,32 @@ suite('closure', () => {
...
+
+  test('fixup array defines', () => {
+    const flags = {
+      jsCode: [{
+        src: '/** @define {boolean} */ var foo = false; console.log(foo)',
+        path: 'foo.js',
+      }],
+      compilationLevel: 'ADVANCED',
+      warningLevel: 'VERBOSE',
+      defines: ['foo=true']
```
Comment (MatrixFrog) [inline test/simple.js]: Let's also add a test with multiple defines (`['foo=true', 'bar=10']` or something)

File: cmd.js [hunk 2]
```diff
@@ -98,12 +98,42 @@ function readFile(path) {
...
+function fixupDefines(flags){
+ /**
+   * compile() expects 'defines' to be object, but minimist may only return array or string
+   * so, we need to convert to proper object.
+   * The supported format will be similar to the following,
+   * https://github.com/google/closure-compiler/wiki/Annotating-JavaScript-for-the-Closure-Compiler#define-type-description
+   * but instead of using --define 'name=value' we should support --defines 'name=value'
+   */
+
+  if (Array.isArray(flags.defines) || typeof flags.defines === "string"){
```
Comment (MatrixFrog) [inline cmd.js]: flags.defines should always be either a string or an array right? So should this `if` have an `else { throw new Error('Invalid value for defines'); }`?

File: test/simple.js (hunk 1 above)
Comment (reshadi) [reply test/simple.js]: ok, will add

File: cmd.js (hunk 2 above)
Comment (reshadi) [reply cmd.js]: it is an optional field. So, it can also be undefined or null. 
This check basically also checks if "flags.defines has value" since those values can only be string or array.
correct?

Comment (MatrixFrog) [reply cmd.js]: Oh of course, I forgot about undefined/null. Let's add a test for those as well (but we should still fail on anything that isn't undefined, null, an array, or a string).

Comment (reshadi) [reply cmd.js]: done

File: cmd.js [hunk 3]
```diff
@@ -98,12 +98,42 @@ function readFile(path) {
...
+    let defines = {};
+    function parseDefines(d) {
```
Comment (MatrixFrog) [inline cmd.js]: nit: This should be called parseDefine since it's only handling one variable each time it's called.

**Comment by MatrixFrog** [comment]
lgtm other than that one nit. thanks!

File: cmd.js (hunk 3 above)
Comment (reshadi) [reply cmd.js]: renamed the function. Also noticed that test is testing the compiler directly and not the cmd.js. Added that as well.

**Comment by MatrixFrog** [comment]
It looks like we need to add some logic for handling quotes.

File: cmd.js [hunk 4]
```diff
@@ -98,14 +98,49 @@ function readFile(path) {
...
+
 /**
  * @param {!Array<{src: string, path: string}>} sources
  * @param {!Array<{src: string, path: string}>} externs
  */
 function ready(sources, externs) {
   const flags = Object.assign(Object.assign({}, argv), {jsCode: sources, externs: externs});
-  const output = compile(flags);
-
+  cmdCompile(flags);
```
Comment (MatrixFrog) [inline cmd.js]: It looks like we still need the 'const output =' since `output` is referenced a few lines down

Comment (reshadi) [reply cmd.js]: oops, sorry! I'm more used to TypeScript IDE catching these error for me. 
I don't have mvn on my machine which is needed to run all tests. If there was a way to use an already released compiler jar for tests in this repo, it would be very helpful.
Sent the patch, hopefully this time passes.
